sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -122,6 +122,14 @@ class SparkSession private(
   val sqlContext: SQLContext = new SQLContext(this)
 
   /**
+   * This is the interface through which the user can get [[JavaSparkContext]]
+   *
+   * @since 2.0.0
+   */
+  @transient
+  val javaSparkContext: JavaSparkContext = new JavaSparkContext(sparkContext)
```

Comment:
Yeah but this exposes a Java-specific API and always creates the object for all callers. It's minor either way, but that seems even less nice.